Service/DataTypeCollector.php

```diff
@@ -14,12 +15,22 @@ class DataTypeCollector
 
     private static $dataTypeList = array();
 
+    private static $keys = ['key', 'label', 'required', 'field_type', 'entity_types'];
+
     /**
      * @param $name
      * @param $class
      */
     public function addDataTypes(DataTypeProviderInterface $dataTypeProvider)
     {
+        foreach ($dataTypeProvider->getDataTypes() as $dataType) {
+            foreach (self::$keys as $key) {
```

Comment:
reduce code complexity by moving this nested foreach to a new protected method.

Reply:
In general:
https://codeclimate.com/github/digipolisgent/symfony_bundle_settings/compare/feature/WEBDOM-283